### PR TITLE
The notifications parameters are now simplified

### PR DESCRIPTION
**Pull request by annando**
The array that is used as a parameter for the `notification` function contained a lot of stuff that wasn't needed at all since it can be fetched from the function itself. The functionality is still a mess, but this is something for another day.

It started as simple change to speed up the display page since the `notify` table doesn't contain an index on the `parent` field.

### CODE REVIEW

File: src/Model/Item.php [hunk 1]
```diff
@@ -1065,9 +1065,6 @@ public static function deleteForUser($condition, $uid)
 			// "Deleting" global items just means hiding them
 			if ($item['uid'] == 0) {
 				DBA::update('user-item', ['hidden' => true], ['iid' => $item['id'], 'uid' => $uid], true);
-
-				// Delete notifications
-				DBA::delete('notify', ['iid' => $item['id'], 'uid' => $uid]);
```
Comment (MrPetovan) [inline src/Model/Item.php:1070]: Why did you remove this code?

File: mod/message.php [hunk 2]
```diff
@@ -282,7 +282,6 @@ function message_content(App $a)
 			$messages = DBA::toArray($messages_stmt);
 
 			DBA::update('mail', ['seen' => 1], ['parent-uri' => $message['parent-uri'], 'uid' => local_user()]);
-			DBA::update('notify', ['seen' => 1], ['type' => Type::MAIL, 'parent' => $message['id'], 'uid' => local_user()]);
```
Comment (MrPetovan) [inline mod/message.php:285]: Why did you remove this code?

File: src/Model/Item.php (hunk 1 above)
Comment (annando) [reply src/Model/Item.php:1070]: We have got a foreign key on this table. So deleting the corresponding `item-uri` will automatically delete this entry as well.

File: mod/message.php (hunk 2 above)
Comment (annando) [reply mod/message.php:285]: I deactivated that this notification type is stored in the `notify` table:
https://github.com/friendica/friendica/blob/818c064c0a5d75b25e1977d336e319e8e9464577/include/enotify.php#L146-L147

They hadn't been displayed anyway in the notifications:
https://github.com/friendica/friendica/blob/develop/mod/ping.php#L414-L424

Unread mails are visible without the use of the `notify` table. And also I had to remove the assigning of the `parent` value since this value isn't referring to the `item` table.

**Review by MrPetovan — APPROVED** [review]
Thank you for the elaboration!
